src/test/java/org/cactoos/text/SplitTextTest.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.text;
+
+import java.io.IOException;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link SplitText}.
+ * @author Fabricio Cabral (fabriciofx@gmail.com)
+ * @version $Id$
+ * @since 0.3
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public final class SplitTextTest {
+
+    @Test
+    public void splitsText() throws IOException {
+        MatcherAssert.assertThat(
+            "Can't split a text",
+            new SplitText(
+                new StringAsText("one two three four five")
+            ),
+            Matchers.contains("one", "two", "three", "four", "five")
```

Comment:
@Timmeey To change `public final SplitText implements Iterable<String>` to `public final SplitText implements Iterable<Text>` we need change the `Matchers.contains("one", "two", "three", "four", "five")` to             `Matchers.contains(new StringAsText("one"), new StringAsText("two"), new StringAsText("three"), new StringAsText("four"),  new StringAsText("five"))`. The problem is to this works, I need implements a method `equals()` into `Text` or `StringAsText`, and a `equals()` method requires an `instanceof` operator, and this projects is against it usage. So, what do you recommend?